contracts/extensions/contracts/src/DevUtils/OrderValidationUtils.sol

```diff
@@ -0,0 +1,173 @@
+            TraderInfo[] memory tradersInfo,
+            bool[] memory isValidSignature
+        )
+    {
+        ordersInfo = EXCHANGE.getOrdersInfo(orders);
+        tradersInfo = getTradersInfo(orders, takerAddresses);
+
+        uint256 length = orders.length;
+        isValidSignature = new bool[](length);
+        for (uint256 i = 0; i != length; i++) {
+            isValidSignature[i] = EXCHANGE.isValidSignature(
+                ordersInfo[i].orderHash,
+                orders[i].makerAddress,
+                signatures[i]
+            );
+        }
+
+        return (ordersInfo, tradersInfo, isValidSignature);
+    }
+
+    /// @dev Fetches balance and allowances for maker and taker of order.
+    /// @param order The order structure.
+    /// @param takerAddress Address that will be filling the order.
+    /// @return Balances and allowances of maker and taker of order.
+    function getTraderInfo(LibOrder.Order memory order, address takerAddress)
+        public
+        view
+        returns (TraderInfo memory traderInfo)
+    {
+        bytes4 makerAssetProxyId = order.makerAssetData.readBytes4(0);
+        bytes4 takerAssetProxyId = order.takerAssetData.readBytes4(0);
+
+        (traderInfo.makerBalance, traderInfo.makerAllowance) = getBalanceAndAllowance(
+            order.makerAddress,
+            EXCHANGE.getAssetProxy(makerAssetProxyId),
+            order.makerAssetData
+        );
+        (traderInfo.takerBalance, traderInfo.takerAllowance) = getBalanceAndAllowance(
+            takerAddress,
+            EXCHANGE.getAssetProxy(takerAssetProxyId),
+            order.takerAssetData
+        );
+        bytes memory zrxAssetData = ZRX_ASSET_DATA;
+        address erc20ProxyAddress = ERC20_PROXY_ADDRESS;
+        (traderInfo.makerZrxBalance, traderInfo.makerZrxAllowance) = getBalanceAndAllowance(
+            order.makerAddress,
+            erc20ProxyAddress,
+            zrxAssetData
+        );
+        (traderInfo.takerZrxBalance, traderInfo.takerZrxAllowance) = getBalanceAndAllowance(
+            takerAddress,
+            erc20ProxyAddress,
+            zrxAssetData
+        );
+        return traderInfo;
+    }
+
+    /// @dev Fetches balances and allowances of maker and taker for each provided order.
+    /// @param orders Array of order specifications.
+    /// @param takerAddresses Array of taker addresses corresponding to each order.
+    /// @return Array of balances and allowances for maker and taker of each order.
+    function getTradersInfo(LibOrder.Order[] memory orders, address[] memory takerAddresses)
+        public
+        view
+        returns (TraderInfo[] memory)
+    {
+        uint256 ordersLength = orders.length;
+        TraderInfo[] memory tradersInfo = new TraderInfo[](ordersLength);
+        for (uint256 i = 0; i != ordersLength; i++) {
+            tradersInfo[i] = getTraderInfo(orders[i], takerAddresses[i]);
+        }
+        return tradersInfo;
+    }
+}
```

Comment:
All I want is a 

```
getOrdersInfo(
        LibOrder.Order[] memory orders,
        bytes[] memory signatures,
        address[] memory takerAddresses
)
```

Method. I don't care about `TradersInfo` once I get back the `remainingFillableAmount`. I guess this refactor can happen in the PR where you add `remainingFillableAmount`.